directord/client.py

```diff
@@ -801,73 +801,87 @@ def run_job(self, lock=None, sentinel=False):
                     _,
                     _,
                 ) = self.driver.job_recv()
-                job = json.loads(data)
-                job["job_id"] = job_id = job.get("job_id", utils.get_uuid())
-                job["job_sha3_224"] = job_sha3_224 = job.get(
-                    "job_sha3_224", utils.object_sha3_224(job)
-                )
-                self.driver.job_send(
-                    msg_id=job_id,
-                    control=self.driver.job_ack,
-                )
-
-                job_parent_id = job.get("parent_id")
-                job_parent_sha3_224 = job.get("parent_sha3_224")
-                self.log.info(
-                    "Item received: parent job UUID [ %s ],"
-                    " parent job sha3_224 [ %s ], job UUID [ %s ],"
-                    " job sha3_224 [ %s ]",
-                    job_parent_id,
-                    job_parent_sha3_224,
-                    job_id,
-                    job_sha3_224,
-                )
-
-                with utils.ClientStatus(
-                    job_id=job_id,
-                    command=command,
-                    ctx=self,
-                ) as c:
-                    if job_parent_id and not self._parent_check(
-                        conn=c, cache=self.cache, job=job
-                    ):
-                        self.q_return.put(
-                            (
-                                None,
-                                None,
-                                False,
-                                "Job omitted, parent failure",
-                                job,
-                                command,
-                                0,
-                                None,
-                            )
-                        )
-                    else:
-                        c.job_state = self.driver.job_processing
-                        component_kwargs = dict(cache=None, job=job)
-                        self.log.debug(
-                            "Queuing component [ %s ], job_id [ %s ]",
-                            command,
-                            job_id,
-                        )
-                        c.info = "task queued"
-                        q_processes.put(
-                            (
-                                component_kwargs,
-                                command,
-                                info,
-                            )
-                        )
+                self.handle_job(command=command, data=data, info=info)
             else:
                 cache_check_time = self.prune_cache(
                     cache_check_time=cache_check_time
                 )
                 time.sleep(poller_interval * 0.001)
 
             if sentinel:
+                self.driver.job_stop()
```

Comment:
this should be `job_close`, unless the messaging driver will use the close method from the abstract elsewhere.

https://github.com/cloudnull/directord/blob/main/directord/drivers/__init__.py#L268

Reply:
Done.